internal/robot/client.go

```diff
@@ -0,0 +1,94 @@
+package robot
+
+import (
+	"time"
+
+	hrobot "github.com/syself/hrobot-go"
+	robotmodels "github.com/syself/hrobot-go/models"
+)
+
+type cacheRobotClient struct {
+	robotClient hrobot.RobotClient
+	timeout     time.Duration
+
+	lastUpdate time.Time
+
+	// cache
+	l []robotmodels.Server
+	m map[int]*robotmodels.Server
+}
+
+func NewClient(robotClient hrobot.RobotClient, cacheTimeout time.Duration) Client {
+	return &cacheRobotClient{
+		timeout:     cacheTimeout,
+		robotClient: robotClient,
+	}
+}
+
+func (c *cacheRobotClient) ServerGet(id int) (*robotmodels.Server, error) {
+	if c.shouldSync() {
+		list, err := c.robotClient.ServerGetList()
+		if err != nil {
+			return nil, err
+		}
+
+		// populate list
+		c.l = list
+
+		// remove all entries from map and populate it freshly
+		c.m = make(map[int]*robotmodels.Server)
+		for i, server := range list {
+			c.m[server.ServerNumber] = &list[i]
+		}
+
+		// set time of last update
+		c.lastUpdate = time.Now()
+	}
```

Comment:
I think this can be deduplicated, the same is in the next function. Maybe move it into shouldSync and rename it  RefreshCache ?

Reply:
Refactored it a bit :)